levels.go

```diff
@@ -425,9 +425,7 @@ func (s *levelsController) compactBuildTables(
 		// from not doing this ASAP after all file creation has finished because this is a
 		// background operation.
 		firstErr = syncDir(s.kv.opt.Dir)
-	}
-
-	if firstErr != nil {
```

Comment:
This `firstErr` can be returned by `syncDir`, if move to `else`, then `syncDir` error is not checked.